changelog/5673.improvement.md

```diff
@@ -0,0 +1,4 @@
+Return diagnostic data for action and NLU predictions.
+
+DIET and TED now both expose the attention weights of their transformer layers.
```

Comment:
@JEM-Mosig I think it would be great if the changelog would document how to retrieve this data (maybe even writing a docs page for it)

Reply:
@erohmensing Where in the docs would this go? Essentially we're adding diagnostic data output to NLU and Core models, which is part of the messages and predictions, respectively. This will be used by Rasa-lit.

Reply:
> which is part of the messages and predictions, respectively.

So like in the parse data? or is it printed in the logs as well? where's this info exposed to the user (if at all)?

Reply:
It is not exposed to the user. For the NLU pipeline, we just add an extra field `diagnostic_data` to the message upon `Component.process` and in Core we add a `diagnostic_data` property to the `Prediction` object. Neither is used by any other component. Its just so that Vincent can build tools to explore what's happening inside our models during inference.

Reply:
Of course, other people could also build such tools by accessing these new properties. But this would not be the default thing to do when building bots.

Reply:
got it. then I would say that the code reference is probably enough, and you can flesh out the changelog and link to the code reference? something like

Add `diagnostic_data` field to the `Message`[link to [here](https://rasa.com/docs/rasa/reference/rasa/shared/nlu/training_data/message#message-objects), i believe] and `Prediction`[link to [here](https://rasa.com/docs/rasa/2.0.x/reference/rasa/core/policies/ensemble#prediction-objects)] objects, which contain information about xyz.

Reply:
Ok, will do that. Thanks!